default/python/AI/ShipDesignAI.py

```diff
@@ -1441,6 +1440,66 @@ def _partclass_in_design(self, partclass):
         """
         return any(part.partClass in partclass for part in self.parts)
 
+    def _calculate_weapon_strength(self, weapon_part, ignore_species=False):
```

Comment:
maybe split to two methods:

_calculate_absolute_weapon_strength(weapon_part)
_calculate_species_weapon_strength(absolute_weapon_strength, species)